pytoda/datasets/protein_sequence_dataset.py

```diff
@@ -34,7 +78,10 @@ def __init__(
         device: torch.device = (
             torch.device('cuda' if torch.cuda.is_available() else 'cpu')
         ),
-        name: str = 'protein-sequences'
+        backend: str = 'eager',
+        iterate_dataset: bool = False,
```

Comment:
any reason why the default is false here but true for smiles datasets?

Reply:
It's False here because the Proteins default vocab is the definite one, and no tokens need be added depending on the dataset and/or transformations.
It takes a lot of time to iterate and the only benefit is the max length, which is often known from preprocessing.